libs-scala/resources-akka/src/test/suite/scala/com/digitalasset/resources/akka/AkkaResourceOwnerSpec.scala

```diff
@@ -35,56 +33,60 @@ final class AkkaResourceOwnerSpec extends AsyncWordSpec with Matchers {
         }
       }
 
-      val resource = for {
-        actorSystem <- Factories
-          .forActorSystem(() => ActorSystem("TestActorSystem"))
-          .acquire()
-        actor <- Factories
-          .successful(actorSystem.actorOf(Props(new TestActor)))
-          .acquire()
+      val resourceOwner = for {
+        actorSystem <- Factories.forActorSystem(() => ActorSystem("TestActorSystem"))
+        actor <- Factories.successful(actorSystem.actorOf(Props(new TestActor)))
       } yield (actorSystem, actor)
 
-      for {
-        resourceFuture <- resource.asFuture
-        (actorSystem, actor) = resourceFuture
-        _ = actor ! 7
-        result <- testPromise.future
-        _ <- resource.release()
-      } yield {
+      val actorSystemReference = new AtomicReference[ActorSystem]()
+
+      val resultFuture = resourceOwner
+        .use { case (actorSystem, actor) =>
+          actorSystemReference.set(actorSystem)
+          actor ! 7
+          testPromise.future
+        }
+
+      blocking {
```

Comment:
If you switch `AnyWordSpec` to `AsyncWordSpec`, you will be able to do this (here and below), as you can return a `Future[Assertion]`.

```scala
    resourceOwner
      .use { (actorSystem, actor) =>
        actor ! 7
        testPromise.future.map { result =>
          result should be(7)
          actorSystem
        } 
      }
      .map { actorSystem =>
        an[IllegalStateException] should be thrownBy actorSystemReference.get.actorOf(
          Props(new TestActor)
        )
      }
```

I find this a lot simpler, as there's no calls to `Await.result` or `AtomicReference#set`.

Reply:
Ah, you're right, `release` is being called earlier, gimme a sec.

Reply:
Done